sources/iTermPreferencesBaseViewController.m

```diff
@@ -270,6 +295,13 @@ - (void)updateValueForInfo:(PreferenceInfo *)info {
             break;
         }
 
+        case kPreferenceInfoTypeUPopup: {
+            assert([info.control isKindOfClass:[NSPopUpButton class]]);
+            NSPopUpButton *popup = (NSPopUpButton *)info.control;
+            [popup selectItemWithTag:[self uintForKey:info.key]];
```

Comment:
Assert that `[self unsignedIntegerForKey:info.key]` will fit in a signed NSInteger (is less than or equal to `NSIntegerMax`), since tags are signed.